src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -78,7 +79,7 @@ public override void Dispose()
 
         protected override bool OnSysCall(uint method)
         {
-            if (!AddGas(InteropService.GetPrice(method, CurrentContext.EvaluationStack)))
+            if (!AddGas(InteropService.GetPrice(method, this)))
```

Comment:
What about we set the price to `0` and call `AddGas()` in `PutEx()`?

Reply:
It's not recommended, it doesn't feel like a uniform operation.